### PR TITLE
fixed copy ctor; fUseGrad setting in MnApplication was lost before

### PR DESCRIPTION
**Pull request by HDembinski**
This patch removes a wrongly implemented copy constructor, which lost the fUseGrad setting (if it was `true` in the original, the copy had `false`), effectively replacing it with the default copy ctor, which is perfectly fine here.

### CODE REVIEW

File: math/minuit2/inc/Minuit2/MnMigrad.h
Comment (eguiraud) [inline math/minuit2/inc/Minuit2/MnMigrad.h:75]:
```suggestion
   MnMigrad(const MnMigrad& migr) = default;
   ~MnMigrad() {}
```

**Review by eguiraud — COMMENTED** [review]
better be explicit about what we want?

**Comment by eguiraud** [comment]
@phsft-bot build please

**Review by lmoneta — APPROVED** [review]
LGTM !
Nice catch for finding this!
I would add the suggestion from Enrico that makes the code more readable

**Comment by lmoneta** [comment]
Thank you for the PR

File: math/minuit2/inc/Minuit2/MnMigrad.h
Comment (HDembinski) [reply math/minuit2/inc/Minuit2/MnMigrad.h:75]: I follow the "rule of zero" here, so no, I think we should not add an explicitly defaulted copy ctor.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c20-if-you-can-avoid-defining-default-operations-do

**Comment by HDembinski** [comment]
> Thank you for the PR

Thank you for approving my PRs :).

File: math/minuit2/inc/Minuit2/MnMigrad.h
Comment (eguiraud) [reply math/minuit2/inc/Minuit2/MnMigrad.h:75]: Not that we _have_ to agree with such guidelines, but anyway right below, this applies in this case: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all as the destructor and the copy-assignment operators are defined.

Comment (HDembinski) [reply math/minuit2/inc/Minuit2/MnMigrad.h:75]: Ok, but making copy assignment private makes no sense here. If the class is default copyable it is also default assignable.

Comment (HDembinski) [reply math/minuit2/inc/Minuit2/MnMigrad.h:75]: I also removed the private assignment, so I think this ready to merge now.

Comment (eguiraud) [reply math/minuit2/inc/Minuit2/MnMigrad.h:75]: this is not going in the direction I suggested xD but I'm not the maintainer of Minuit, so up to @lmoneta , I don't have a strong opinion

**Review by lmoneta — APPROVED** [review]
I agree that there is no reason to have the assignment operator disabled for the class.
The PR seems ready to be merged

**Comment by lmoneta** [comment]
Actually investigating in more detail I think the reason that the assignment operator is private is because the base class MnApplication has a data member a const reference to the FCNBase object and it cannot be copied.

**Comment by HDembinski** [comment]
@lmoneta Just for the record, since I was mistaken and @eguiraud was right, I am not against making the copy ctor explicitly defaulted and the assignment operator explicitly deleted (with some inline comment why)

**Comment by HDembinski** [comment]
@lmoneta I resolved the discussion with @eguiraud so I think this can be merged now, right?

**Comment by lmoneta** [comment]
Hi, yes it is fine now. Let me run the CI again and then I will merge. Thank you for the update

**Comment by lmoneta** [comment]
@phsft-bot  build !

**Pull request merged**
